docs/concepts/index.md

```diff
@@ -4,10 +4,12 @@ nav_order: 3
 ---
 # Core Concepts
 
-The framework is broken up into three core concepts: Generators, Actions, and Agents.
+The Sublayer framework is structured around three primary components: Generators, Actions, and Agents. These components work together to power the framework's LLM-based applications.
 
-Browse the links below to go more in depth into each of these concepts:
+- **Generators**: These are responsible for producing specific outputs from given inputs. They handle a single generation task without engaging in complex decision-making. [Learn more about Generators]({% link docs/concepts/generators.md %}).
 
-* [Generators]({% link docs/concepts/generators.md %})
-* [Actions]({% link docs/concepts/actions.md %})
-* [Agents]({% link docs/concepts/agents.md %})
+- **Actions**: Actions carry out specific tasks, such as obtaining inputs for a Generator or using outputs from a Generator. They focus on executing single, defined operations without complex decision-making. [Discover more about Actions]({% link docs/concepts/actions.md %}).
+
+- **Agents**: These act like personal assistants that respond to repetitive tasks or situational changes. By monitoring triggers, evaluating goals, checking status, and executing steps, Agents automate processes efficiently. [Understand more about Agents]({% link docs/concepts/agents.md %}).
+
+Explore the links above for a deeper dive into each concept. Each is integral to creating robust, scalable solutions with Sublayer.
```

Comment:
Im not sure about this file. 

on one hand too verbose...
But on the other, still an improvement from how it was.